torch/_refs/__init__.py

```diff
@@ -378,7 +378,7 @@ def _isnan(a: Tensor) -> Tensor:
 square = _make_elementwise_unary_reference(
     prims.square,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.BOOL_TO_LONG,
-    aten_op=None,  # CompositeImplicitAutograd
+    aten_op=torch.ops.aten.square.out,
```

Comment:
why do we need `out` overload here?

Reply:
It's composite implicit autograd but I guess we didn't do the out variant correctly, not sure

Reply:
```
- func: square(Tensor self) -> Tensor
  device_check: NoCheck   # TensorIterator
  variants: function, method

- func: square_(Tensor(a!) self) -> Tensor(a!)
  device_check: NoCheck   # TensorIterator
  variants: function, method

- func: square.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
  dispatch:
    CPU, CUDA: square_out
    MPS: square_out_mps
    ```

TBH this is kind of weird.

Reply:
Yeah, CPU, CUDA dispatch for `out` is wrong, it calls `return at::pow_out(result, self, 2);`, should still be composite implicit autograd.